pkg/helpers/identifier_string_parser.go

```diff
@@ -25,8 +24,126 @@ func parseIdentifierStringWithOpts(identifier string, opts func(*csv.Reader)) ([
 	if err != nil {
 		return nil, fmt.Errorf("unable to read identifier: %s, err = %w", identifier, err)
 	}
+	if lines == nil {
+		return make([]string, 0), nil
+	}
 	if len(lines) != 1 {
 		return nil, fmt.Errorf("incompatible identifier: %s", identifier)
 	}
 	return lines[0], nil
 }
+
+func ParseIdentifierString(identifier string) ([]string, error) {
+	return parseIdentifierStringWithOpts(identifier, func(r *csv.Reader) {
+		r.Comma = IdDelimiter
+	})
+}
+
+func ParseResourceIdentifier(identifier string) []string {
+	if identifier == "" {
+		return make([]string, 0)
+	}
+	return strings.Split(identifier, string(ResourceIdDelimiter))
+}
+
+func EncodeResourceIdentifier(parts ...string) string {
+	return strings.Join(parts, string(ResourceIdDelimiter))
+}
+
+func parseAccountObjectIdentifier(identifier string, parser identifierParsingFunc) (sdk.AccountObjectIdentifier, error) {
```

Comment:
all this unexported functions are almost the same, can't we just have a parametrized func like:
`func parseIdentifier[I sdk.ObjectIdentifier](identifier string, parser (however it's always ParseIdentifierString currently , so maybe we can inline it), ex[pectedParts int, expectedFormat string(like "<account_object_name>"))`? it will be much less code and will be much more readable, because now one may wonder what happens in these functions if each of them is rewritten

Reply:
Right, changed